stdlib-meta-update-imports.patch

```diff
@@ -1,27 +1,852 @@
++open import Reflection.AST.Term
 +import Reflection.AST.Abstraction as Abs
 +open import Reflection.AST.Argument as Arg hiding (map)
++import Reflection.AST.Name as Name
++open import Reflection.AST.Pattern
 +open import Reflection.AST.Argument.Information
 +open import Reflection.AST.Argument.Visibility as Vis
 +open import Reflection.TCM.Syntax
  
  open import Data.Unit
- open import Data.Nat
+ open import Data.Product hiding (map; zip)
+diff --git a/Interface/DecEq.agda b/Interface/DecEq.agda
+index 366e594..3322dc8 100644
+--- a/Interface/DecEq.agda
++++ b/Interface/DecEq.agda
+@@ -1,6 +1,6 @@
+ {-# OPTIONS --safe --without-K #-}
+ 
+-open import Prelude
++open import MetaPrelude
+ 
+ open import Relation.Binary
+ 
+diff --git a/Interface/Monad.agda b/Interface/Monad.agda
+index 0172b00..8c97e20 100644
+--- a/Interface/Monad.agda
++++ b/Interface/Monad.agda
+@@ -2,7 +2,7 @@
+ 
+ module Interface.Monad where
+ 
+-open import Prelude
++open import MetaPrelude
+ 
+ private
+   variable
+diff --git a/Interface/MonadError.agda b/Interface/MonadError.agda
+index d172454..4c90c47 100644
+--- a/Interface/MonadError.agda
++++ b/Interface/MonadError.agda
+@@ -2,7 +2,7 @@
+ 
+ open import Level
+ 
+-open import Prelude
++open import MetaPrelude
+ 
+ open import Interface.Monad
+ open import Reflection using (TC; ErrorPart; typeError; catchTC; strErr)
+diff --git a/Interface/MonadReader.agda b/Interface/MonadReader.agda
+index 12f4c3e..e4e1ffc 100644
+--- a/Interface/MonadReader.agda
++++ b/Interface/MonadReader.agda
+@@ -2,7 +2,7 @@
+ 
+ module Interface.MonadReader where
+ 
+-open import Prelude
++open import MetaPrelude
+ 
+ open import Interface.Monad
+ open import Interface.MonadError
+diff --git a/Interface/MonadTC.agda b/Interface/MonadTC.agda
+index 677963f..812bd58 100644
+--- a/Interface/MonadTC.agda
++++ b/Interface/MonadTC.agda
+@@ -2,7 +2,7 @@
+ 
+ module Interface.MonadTC where
+ 
+-open import Prelude
++open import MetaPrelude
+ 
+ open import Data.List using (map)
+ 
+diff --git a/Lenses/Derive.agda b/Lenses/Derive.agda
+index 933dee4..31d563c 100644
+--- a/Lenses/Derive.agda
++++ b/Lenses/Derive.agda
+@@ -11,7 +11,7 @@ open import Data.Nat using (ℕ)
+ open import Relation.Binary.PropositionalEquality using (_≡_; refl)
+ 
+ open import Reflection hiding (_>>=_; _>>_)
+-open import Reflection.Meta
++open import Reflection.AST.Meta
+ 
+ open import Generics
+ open Debug ("DeriveLenses" , 100)
+diff --git a/Main.agda b/Main.agda
+index b8dbffc..0cd0ba2 100644
+--- a/Main.agda
++++ b/Main.agda
+@@ -1,11 +1,21 @@
+--- {-# OPTIONS --safe --without-K #-}
++{-# OPTIONS --safe #-}
+ module Main where
+ 
+-open import Generics
+-open import Classes
+-open import Lenses
+-open import Tactics
++-- open import Generics
++-- open import Classes
++-- open import Lenses
++-- open import Tactics
+ 
+--- TODO: incorporate into the rest
+-module Stash where
+-  open import Everything
++-- -- TODO: incorporate into the rest
```

Comment:
I'm not clear on what's left to do here, but I trust you know what you're doing and can decide whether or not to merge this possibly without first completing the todo items.

Reply:
This is really some cleanup that should be done in stdlib-meta at some point. That library needs some major cleanup in general, since it contains a bunch of copies of incompatible stuff.

Reply:
got it. thanks.